core/src/main/java/io/aiven/kafka/tieredstorage/metrics/Metrics.java

```diff
@@ -32,107 +33,267 @@
+    private void recordSegmentCopyRequests(final TopicPartition topicPartition) {
+        new SensorProvider(metrics, sensorName(SEGMENT_COPY))
+            .with(metricsRegistry.segmentCopyRequestsRate, new Rate())
+            .with(metricsRegistry.segmentCopyRequestsTotal, new CumulativeCount())
+            .get()
+            .record();
+        new SensorProvider(metrics, sensorNameByTopic(topicPartition, SEGMENT_COPY),
+            () -> topicTags(topicPartition))
+            .with(metricsRegistry.segmentCopyRequestsRateByTopic, new Rate())
+            .with(metricsRegistry.segmentCopyRequestsTotalByTopic, new CumulativeCount())
+            .get()
+            .record();
+        new SensorProvider(metrics, sensorNameByTopicPartition(topicPartition, SEGMENT_COPY),
+            () -> topicPartitionTags(topicPartition), Sensor.RecordingLevel.DEBUG)
+            .with(metricsRegistry.segmentCopyRequestsRateByTopicPartition, new Rate())
+            .with(metricsRegistry.segmentCopyRequestsTotalByTopicPartition, new CumulativeCount())
+            .get()
+            .record();
+    }
 
-        segmentDeleteBytes = metrics.sensor("segment-delete-bytes");
-        segmentDeleteBytes.add(metrics.metricName("segment-delete-bytes-rate", metricGroup), new Rate());
-        segmentDeleteBytes.add(metrics.metricName("segment-delete-bytes-total", metricGroup), new CumulativeSum());
+    public void recordSegmentCopyTime(final TopicPartition topicPartition, final long startMs, final long endMs) {
+        final var time = endMs - startMs;
+        new SensorProvider(metrics, sensorName(SEGMENT_COPY_TIME))
+            .with(metricsRegistry.segmentCopyTimeAvg, new Avg())
+            .with(metricsRegistry.segmentCopyTimeMax, new Max())
+            .get()
+            .record(time);
+        new SensorProvider(metrics, sensorNameByTopic(topicPartition, SEGMENT_COPY_TIME),
+            () -> topicTags(topicPartition))
+            .with(metricsRegistry.segmentCopyTimeAvgByTopic, new Avg())
+            .with(metricsRegistry.segmentCopyTimeMaxByTopic, new Max())
+            .get()
+            .record(time);
+        new SensorProvider(metrics, sensorNameByTopicPartition(topicPartition, SEGMENT_COPY_TIME),
+            () -> topicPartitionTags(topicPartition), Sensor.RecordingLevel.DEBUG)
+            .with(metricsRegistry.segmentCopyTimeAvgByTopicPartition, new Avg())
+            .with(metricsRegistry.segmentCopyTimeMaxByTopicPartition, new Max())
+            .get()
+            .record(time);
+    }
 
-        segmentDeleteTime = metrics.sensor("segment-delete-time");
-        segmentDeleteTime.add(metrics.metricName("segment-delete-time-avg", metricGroup), new Avg());
-        segmentDeleteTime.add(metrics.metricName("segment-delete-time-max", metricGroup), new Max());
+    public void recordSegmentCopyError(final TopicPartition topicPartition) {
+        recordSegmentCopyErrors(topicPartition);
+    }
```

Comment:
Do we need this function or `recordSegmentCopyErrors` can be just made public?

Reply:
Sure, removing inner method.